plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java

```diff
@@ -510,6 +510,21 @@ public boolean isDropTableDangerous() {
             ((Drop) statement).getName() != null && ((Drop) statement).getType().equalsIgnoreCase("table");
     }
 
+    public boolean isModifying() {
+        if (getType() == SQLQueryType.UNKNOWN) {
+            return false;
+        }
+        if (statement instanceof Select) {
+            SelectBody selectBody = ((Select) statement).getSelectBody();
+            if (selectBody instanceof PlainSelect plainSelectBody) {
+                return plainSelectBody.isForUpdate() || plainSelectBody.getIntoTables() != null;
+            }
+            return false;
+        } else {
+            return true;

```

Comment:
No sure we can use this approach. There are many non-SELECT queries which are not modifying. E.g. USE, SHOW, WITH, etc. Every dialect may have extra allowed statements.
Let's restrict only INSERT, DELETE, UPDATE, CREATE, DROP, ALTER for now.